elasticsearch-odfe/plan.sh

```diff
@@ -0,0 +1,96 @@
+ELASTICSEARCH_VERSION="6.5.4"
+ELASTICSEARCH_PKG_URL="https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-$ELASTICSEARCH_VERSION.tar.gz"
+pkg_version=0.7.0.1
+pkg_name="elasticsearch-odfe"
+pkg_description="Open Distro for Elasticsearch plugins"
+pkg_origin="core"
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=("Revised BSD, Apache 2.0")
+pkg_upstream_url="https://github.com/opendistro-for-elasticsearch"
+pkg_build_deps=(
+  core/coreutils
+  core/git
+  core/maven
+  core/openjdk11
+  core/openssl
+  core/zip
+)
+pkg_deps=(
+  core/coreutils
+  core/curl
+  core/glibc
+  core/openjdk11
+  core/procps-ng
+  core/ruby
+  core/wget
+  core/zlib
+)
+
+pkg_bin_dirs=(bin)
+pkg_binds_optional=(
+  [elasticsearch]="http-port transport-port"
+)
+pkg_lib_dirs=(lib)
+pkg_exports=(
+  [http-port]=es_yaml.http.port
+  [transport-port]=es_yaml.transport.tcp.port
+)
+pkg_exposes=(http-port transport-port)
+
+ODFE_DEPENDENCIES=(
+  'security-parent'
+  'security-ssl'
+  'security-advanced-modules'
+)
+
+do_download() {
+  wget -O "${HAB_CACHE_SRC_PATH}/elasticsearch-oss-${ELASTICSEARCH_VERSION}.tar.gz" "${ELASTICSEARCH_PKG_URL}"
+  for component in "${ODFE_DEPENDENCIES[@]}"; do
```

Comment:
In general, we try to ship software with the upstream defaults in the core origin. Plugins can make this a bit problematic. IIRC the ssl plugins was one of the big deals for opendistro, but I'd like to confirm that the ones we're shipping are the defaults, or expected as baseline functionality.

Reply:
This is a really good callout, and there are more defaults I think we should ship with. Here's the list:
```
opendistro-sql
opendistro-alerting
opendistro-security
performance-analyzer

At the moment we are shipping just the security plugins, so I'll have to add the rest.

Reply:
After looking deeper at the remaining plugins, I've determined that these plugins are intended to be built and installed in a standalone manner on top of the build security+elasticsearch package:

```
opendistro-sql
opendistro-alerting
opendistro-security
performance-analyzer
```

I think the correct approach with these plugins is to build them as separate packages, and then have a user dep on them in their configuration package.

I think I could build these packages, but I think it's out of scope for this PR.

Reply:
I would agree with that.  Let's keep this this simple and as close to upstream's intent as possible. If I'm following along correctly, that means the above 4 plugins should be added by a user at their own discretion.